holoviews/tests/plotting/plotly/test_barplot.py

```diff
@@ -91,3 +92,63 @@ def test_visible(self):
         element = Bars([3, 2, 1]).opts(visible=False)
         state = self._get_plot_state(element)
         self.assertEqual(state['data'][0]['visible'], False)
+    def test_bars_continuous_data_list_same_interval(self):
+        bars = Bars(([0, 1, 2], [10, 20, 30]))
+        plot = self._get_plot_state(bars)
+        self.assertEqual(plot['data'][0]['x'], [0, 1, 2])
+        self.assertEqual(plot['data'][0]['y'], [10, 20, 30])
+
+    def test_bars_continuous_data_list_diff_interval(self):
+        bars = Bars(([0, 3, 10], [10, 20, 30]))
+        plot = self._get_plot_state(bars)
+        self.assertEqual(plot['data'][0]['x'], [0, 3, 10])
+        self.assertEqual(plot['data'][0]['y'], [10, 20, 30])
+
+    def test_bars_continuous_datetime(self):
+        y = np.random.rand(10)
+        bars = Bars((pd.date_range("1/1/2000", periods=10), y))
+        plot = self._get_plot_state(bars)
+        self.assertEqual(plot['data'][0]['x'], pd.date_range("1/1/2000", periods=10).values.astype(float))
+        self.assertEqual(plot['data'][0]['y'], y)
+
+    def test_bars_not_continuous_data_list(self):
+        bars = Bars([("A", 1), ("B", 2), ("C", 3)])
+        plot = self._get_plot_state(bars)
+        self.assertEqual(plot['data'][0]['x'], ["A", "B", "C"])
+        self.assertEqual(plot['data'][0]['y'], [1, 2, 3])
+
+    def test_bars_group(self):
+        samples = 100
+
+        pets = ["Cat", "Dog", "Hamster", "Rabbit"]
+        genders = ["Female", "Male", "N/A"]
+
+        np.random.seed(100)
+        pets_sample = np.random.choice(pets, samples)
+        gender_sample = np.random.choice(genders, samples)
+
+        bars = Bars(
+            (pets_sample, gender_sample, np.ones(samples)), ["Pets", "Gender"]
+        ).aggregate(function=np.sum)
+        plot = self._get_plot_state(bars)
+        self.assertEqual(set(plot['data'][0]['x'][0]), set(pets))
```

Comment:
Is order unimportant, since you use sets?

I would think order matter for these kind of plots.

Reply:
I use set because there are duplicate values since it's multi-categorical, e.g. Dog Dog Dog, etc.